src/file/FileControllerNative.ts

```diff
@@ -24,30 +25,62 @@ export class FileControllerNative implements FileController {
 	 * Temporary files are deleted afterwards in apps.
 	 */
 	async download(file: TutanotaFile) {
-		await guiDownload(this.doDownload(file))
+		await guiDownload(
+			isAndroidApp() || isDesktop()
+				? this.doDownload(
+						new Array(file),
+						(file) => this.downloadAndDecryptInNative(file),
+						(files) => promiseMap(files, (file) => this.fileApp.putFileIntoDownloadsFolder(file.location)),
+						(files) => this.cleanUp(files),
+				  )
+				: this.doDownload(
+						new Array(file),
+						(file) => this.downloadAndDecryptInNative(file),
+						(files) => promiseMap(files, (file) => this.fileApp.open(file)),
+						(files) => this.cleanUp(files),
+				  ),
+		)
 	}
 
 	async open(file: TutanotaFile) {
 		await guiDownload(this.doOpen(file))
 	}
 
-	private async doDownload(tutanotaFile: TutanotaFile) {
-		let temporaryFile: FileReference | null = null
-		try {
-			temporaryFile = await this.downloadAndDecryptInNative(tutanotaFile)
-			if (isAndroidApp() || isDesktop()) {
-				await this.fileApp.putFileIntoDownloadsFolder(temporaryFile.location)
-			} else {
-				await this.fileApp.open(temporaryFile)
+	private async cleanUp(files: FileReference[]) {
+		if (files.length > 0) {
+			for (const file of files) {
+				try {
+					await this.fileApp.deleteFile(file.location)
+				} catch (e) {
+					console.log("failed to delete file", file.location, e)
+				}
 			}
-		} finally {
-			if (temporaryFile) {
+		}
+	}
+
+	private async doDownload<T>(
+		tutanotaFiles: TutanotaFile[],
```

Comment:
maybe we should make these arguments an object, it's easier to read this way

Reply:
that's true